server/pbench/bin/pbench-server-prep-version-001

```diff
@@ -0,0 +1,182 @@
+    exit 1
+fi
+
+if [[ "$duplicates" == "" ]] ;then
+    echo "Failed: \"getconf.py duplicates-dir pbench-server" $errlog
+    exit 1
+fi
+
+log_init $(basename $0) $LOGSDIR/$(basename $0)
+
+tmp=$TMP/$PROG.$$
+
+function quarantine () {
+    dest=$1
+    files="$@"
+
+    mkdir -o $dest || return $?
+    mv $files $dest || return $?
+}
+
+trap 'rm -rf $tmp' EXIT INT QUIT
+
+mkdir -p $tmp
+list=$tmp/list.check
+status=$tmp/status
+> $status
+results_dir=$(getconf.py receive-dir pbench-server)
+# check for results that are ready for processing
+find ${results_dir} -maxdepth 2 -name *.tar.xz.md5 > $list
+
+typeset -i ntbs=0
+typeset -i nerrs=0
+typeset -i nquarantined=0
+typeset -i ndups=0
+
+echo $TS
+while read ck ;do
+    # full pathname of tarball
+    tb=${ck%.md5}
+
+    # directory
+    tbdir=$(dirname ${tb})
+
+    # resultname: get the basename foo.tar.xz and then strip the .tar.xz
+    resultname=$(basename ${tb})
+    resultname=${resultname%.tar.xz}
+
+    # hostname is the last component of the directory part of the full path
+    controller=$(basename ${tbdir})
+
+    dest=$archive/$controller
+
+    # if there is a prefix file with the old name, normalize it now
+    prefixname=${tbdir}/prefix.${resultname}
+    if [ -f ${prefixname} ] ;then
+        mv ${prefixname} ${tbdir}/${resultname}.prefix
+        sts=$?
+        if [ $sts -ne 0 ] ;then
+            echo "$TS: Failed: \"mv ${prefixname} ${tbdir}/${resultname}.prefix\", status $sts" >> $status
+            # record it but do not fail: we can always fix the prefix after the fact.
+        else
+            prefixname=${tbdir}/${resultname}.prefix
+        fi
+    fi
+
+    # sanity checks
+    if [ ! -f ${tb} ] ;then
+        echo "$TS: ${tb} file does not exist" >&4
+        quarantine $quarantine/$controller ${tb}.md5 $prefixname
+        sts=$?
+        if ...
+        nerrs=$nerrs+1
+        conntinue
+    fi
+    # Impossible: the md5 file disappeared between the find and this check>
+    if [ ! -f ${tb}.md5 ] ;then
+        echo "$TS: ${tb}.md5 file does not exist" >&4
+        quarantine $quarantine/$controller ${tb} $prefixname
+        sts=$?
+        if ...
+        nerrs=$nerrs+1
+        continue
+    fi
+
+    if [ -f ${dest}/${resultname}.tar.xz -o -f ${dest}/${resultname}.tar.xz.md5 ] ;then
+        quarantine $duplicates/$controller  ${tb} ${tb}.md5 $prefixname
+        ndups=$ndups+1
+    fi
+
+    mkdir -p ${dest}
+    sts=$?
+    if [ $sts -ne 0 ] ;then
+        echo "$TS: Failed: \"mkdir -p ${dest}\", status $sts" >> $status
+        nerrs=$nerrs+1
+        # this is fatal
+        break
+    fi
+
+    mv ${tb} ${tb}.md5 $dest
+    sts=$?
+    if [ $sts -ne 0 ] ;then
+        echo "$TS: Failed: \"mv ${tb} ${tb}.md5 $dest\", status $sts" >> $status
+        nerrs=$nerrs+1
+    fi
+    if [ $sts -eq 0 ] ;then
+        #if the mv succeeded, then delete the check file.
+        # otherwise, try again next time.
+        rm -f ${tb}.check
```

Comment:
Delete the check file?  Not sure what that means...

Reply:
Sorry - the last 3 comments refer to outdated code. I'll push the current code.